### ISSUE TITLE
BrowserAuthError: monitor_window_timeout:

### ISSUE THREAD

**Issue by 01010100010100100100100101000111** [AzureAD/microsoft-authentication-library-for-js#2293]
## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [x] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description
I am using the login redirect flow with Azure B2C (localStorage) and all seems to be working OK, however when I leave the session and come back a day or so later, when it tries to acquireTokenSilent, I get an error of: "BrowserAuthError: monitor_window_timeout: Token acquisition in popup failed due to timeout."

I am not sure if it's because I am doing something wrong / missed some info or if it is a bug, can anyone help? I see this issue has been raised before on MSAL 1.x but it talked about having an empty auth.html page as the redirect and that does not seem like it would work for me.

Edit:

I can also see the below error but again not sure if I should be handling this or not:

    error	"invalid_grant"
    error_description	"AADB2C90080: The provided grant has expired. Please re-authenticate and try again.

**Comment by jmckennon** [comment]
> I am using the login redirect flow with Azure B2C (localStorage) and all seems to be working OK, however when I leave the session and come back a day or so later, when it tries to acquireTokenSilent, I get an error of: "BrowserAuthError: monitor_window_timeout: Token acquisition in popup failed due to timeout."

Sessions lifetimes only last for 24 hours, so if you call `acquireTokenSilent` after 24 hours (and it doesn't have a login function to fall back to) it will error out. This is represented by your `invalid_grant` error you're seeing. I would not expect to see the `"BrowserAuthError: monitor_window_timeout: Token acquisition in popup failed due to timeout."` error, so I'll look into this to see if there actually could be a timeout happening here, or if this error happens to be very uninformative of the actual problem in this case.

**Comment by 01010100010100100100100101000111** [comment]
> 
> 
> > I am using the login redirect flow with Azure B2C (localStorage) and all seems to be working OK, however when I leave the session and come back a day or so later, when it tries to acquireTokenSilent, I get an error of: "BrowserAuthError: monitor_window_timeout: Token acquisition in popup failed due to timeout."
> 
> Sessions lifetimes only last for 24 hours, so if you call `acquireTokenSilent` after 24 hours (and it doesn't have a login function to fall back to) it will error out. This is represented by your `invalid_grant` error you're seeing. I would not expect to see the `"BrowserAuthError: monitor_window_timeout: Token acquisition in popup failed due to timeout."` error, so I'll look into this to see if there actually could be a timeout happening here, or if this error happens to be very uninformative of the actual problem in this case.

Thanks for the quick response. So when you say "it doesn't have a login function to fall back to", what does that mean? 

I am trying to integrate this into a VueJs project and below is my token method so far (work in progress):

```
  public async getAccessTokenAsync(): Promise<string> {
    const token = await this.app
      .acquireTokenSilent({
        scopes: [AuthService.scope],
        account: this.account,
      } as SilentRequest)
      .then(authResponse => {
        return authResponse.accessToken;
      })
      .catch((error: any) => {
        if (error instanceof InteractionRequiredAuthError) {
          this.loginRedirect();
        } else {
          throw error;
        }
        return null;
      });

    if (token == null) {
      throw new Error("Unable to obtain an Access Token");
    }

    return token;
  }
```

**Comment by jmckennon** [comment]
What you have here is actually exactly what my comment suggests--you have a `.catch` after the `acquireTokenSilent` that contains some logic to handle an error, which calls `loginRedirect`.

Can you try setting `loadFrameTimeout` to a higher number, maybe 30 seconds? And report back if you see any changes.

Are you only ever seeing the timeout when you need to establish a session again? Can you take a look at some network traces to see where the bottleneck may be if a timeout is occurring? You can send fiddler traces to the email on my profile.

**Comment by 01010100010100100100100101000111** [comment]
> Are you only ever seeing the timeout when you need to establish a session again?

Yes, it's otherwise working fine. 

Let me try your suggestions on loadFrameTimeout and redirect url to see if that is any different, if not I will have a go at providing some traces (may be a few days until I can that though).

**Comment by 01010100010100100100100101000111** [comment]
I tried this again without any changes and on both localhost and on the web I am now getting invalid_redirect_uri - this was due to it trying to go to "https://localhost:44316/Account" (the page I was logging in from) but on each the app.loginRedirect's I set the redirectUrl to window.location.origin (so yes I incur an extra re-direct, that's fine) but it shouldn't have been trying to re-direct me back to "/Account".

So I added "/Account" to the B2C Config to test and it then (with the expired session) asked me to log-in again. I noticed I don't set a redirectUrl on the the MSAL Config when creating the app, so wondering if due to that being blank, it was using the window location instead?

I have set the redirectUrl on the app's config and re-published it so going to let the session expire again and see if that fixes it, bizarre how it's suddenly a different error.

I will report back on what happens with the above and if I get the original error again, I will try your suggested steps.
